test/unit/test_tss2_rc.c

```diff
@@ -0,0 +1,271 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <setjmp.h>
+#include <cmocka.h>
+
+#include "tss2_rc.h"
+
+#define ARRAY_LEN(x) (sizeof(x)/sizeof(x[0]))
+
+#define TPM2_ERROR_TSS2_RC_LAYER_COUNT (TSS2_RC_LAYER_MASK >> TSS2_RC_LAYER_SHIFT)
+
+#define assert_string_prefix(str, prefix) \
+    assert_memory_equal(str, prefix, strlen(prefix))
+
+static void test_layers(void **state) {
+    (void) state;
+
+    static const char *known_layers[TPM2_ERROR_TSS2_RC_LAYER_COUNT] = {
+        "tpm:",
+        NULL,
+        NULL,
+        NULL,
+        NULL,
+        NULL,
+        "fapi:",
+        "esapi:",
+        "sys:",
+        "mu:",
+        "tcti:",
+        "rmt",
+        "rm",
+        "drvr",
+    };
+
+    UINT8 layer;
+    for (layer = 0; layer < TPM2_ERROR_TSS2_RC_LAYER_COUNT; layer++) {
+        TSS2_RC rc = TSS2_RC_LAYER(layer);
+
+        const char *got = Tss2_RC_Decode(rc);
+
+        char buf[256];
+        snprintf(buf, sizeof(buf), "%u:", layer);
+
+        const char *expected = known_layers[layer] ? known_layers[layer] : buf;
+        assert_string_prefix(got, expected);
+    }
+}
+
+static void test_tpm_format_0_version2_0_error(void **state) {
```

Comment:
Sorry, still wrong coding style...
@williamcroberts

Reply:
Whoops forgot about the test code.

Reply:
Done.